client/client.go

```diff
@@ -442,9 +443,14 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 		return nil, nil, err
 	}
 
+	isWait := false
+	if req != nil && req.URL != nil {
+		isWait, _ = strconv.ParseBool(req.URL.Query().Get("wait"))
+	}
+
 	var hctx context.Context
 	var hcancel context.CancelFunc
-	if c.headerTimeout > 0 {
+	if !isWait && c.headerTimeout > 0 {
```

Comment:
@xiang90 It copies header https://github.com/coreos/etcd/blob/master/proxy/reverse.go#L89 here, but do you think it does something else wrong?